# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/webscheduling/WebTaskScheduler.cpp
# Commit: 78181c32bdc6
# Full Hash: 78181c32bdc6b1d66eac04f3ce11a2902f477823
# Author: Sean Feng <sefeng@mozilla.com>
# Date: 2025-02-27 21:52:52
# Regressor Bug: 1920115
# File Overlap Count: 2
# Description:
#   Bug 1920115 - Add the setup for WebTaskSchedulingState r=jjaschke,dom-core,arai
#   
#   Spec: https://wicg.github.io/scheduling-apis/#scheduling-state
#   
#   This patch add the setup for WebTaskSchedulingState so that it
# ==============================================================================

diff -r 912cf1adac80 -r 78181c32bdc6 dom/webscheduling/WebTaskScheduler.cpp
--- a/dom/webscheduling/WebTaskScheduler.cpp	Thu Feb 27 16:17:34 2025 +0000
+++ b/dom/webscheduling/WebTaskScheduler.cpp	Thu Feb 27 16:17:35 2025 +0000
@@ -22,6 +22,15 @@
   ImplCycleCollectionTraverse(aCallback, aQueue.Tasks(), aName, aFlags);
 }
 
+NS_IMPL_CYCLE_COLLECTION_CLASS(WebTaskSchedulingState)
+
+NS_IMPL_CYCLE_COLLECTION_TRAVERSE_BEGIN(WebTaskSchedulingState)
+  NS_IMPL_CYCLE_COLLECTION_TRAVERSE(mAbortSource, mPrioritySource);
+NS_IMPL_CYCLE_COLLECTION_TRAVERSE_END
+
+NS_IMPL_CYCLE_COLLECTION_UNLINK_BEGIN(WebTaskSchedulingState)
+  NS_IMPL_CYCLE_COLLECTION_UNLINK(mAbortSource, mPrioritySource);
+NS_IMPL_CYCLE_COLLECTION_UNLINK_END
 NS_IMPL_CYCLE_COLLECTION_CLASS(WebTask)
 
 NS_IMPL_CYCLE_COLLECTION_TRAVERSE_BEGIN(WebTask)